changelog.d/10485.bugfix

```diff
@@ -0,0 +1 @@
+Fix a long-standing bug where Synapse would signal "no change" in device one-time key counts, whereas in reality the number of keys had been exhausted.
```

Comment:
Feedback on whether this is digestible enough is welcome.

Reply:
Maybe

> Fix a long-standing bug where Synapse would mistakenly tell clients one-time key counts hadn't changed instead of telling them no key is available.

Would be easier to read?

Reply:
I think it would be nice to express this in terms of user-visible impact. 
```suggestion
Fix a long-standing bug where Synapse would not inform clients that a device had exhausted its one-time-key pool, potentially causing problems decrypting events.
```

Reply:
Ah yeah that sounds better

Reply:
Thanks both!